linkerd/app/outbound/src/tests.rs

```diff
@@ -89,9 +95,320 @@ async fn plaintext_tcp() {
 
     // Build the outbound TCP balancer stack.
     let outbound_tcp = cfg.build_tcp_balance(&connect, resolver, prevent_loop, &metrics.outbound);
+    // Oneshot the TCP stack with the target and request (client conn).
+    outbound_tcp.make_oneshot(target_addr, client_io).await
```

Comment:
Does this test need to assert that the client/server read/write what they expect?

Reply:
the mock IO types do this for us automatically --- they'll panic if they see unexpected `read`/`write` calls, or if they're dropped without completing their "script"s